pkg/sdk/authentication_policies_def.go

```diff
@@ -0,0 +1,126 @@
+				Name().
+				OptionalQueryStructField(
+					"Set",
+					g.NewQueryStruct("AuthenticationPolicySet").
+						ListAssignment("AUTHENTICATION_METHODS", "AuthenticationMethods", g.ParameterOptions().Parentheses()).
+						ListAssignment("MFA_AUTHENTICATION_METHODS", "MfaAuthenticationMethods", g.ParameterOptions().Parentheses()).
+						OptionalTextAssignment("MFA_ENROLLMENT", g.ParameterOptions().SingleQuotes()).
+						ListAssignment("CLIENT_TYPES", "ClientTypes", g.ParameterOptions().Parentheses()).
+						ListAssignment("SECURITY_INTEGRATIONS", "SecurityIntegrationsOption", g.ParameterOptions().Parentheses()).
+						OptionalTextAssignment("COMMENT", g.ParameterOptions().SingleQuotes()).
+						WithValidation(g.AtLeastOneValueSet, "AuthenticationMethods", "MfaAuthenticationMethods", "MfaEnrollment", "ClientTypes", "SecurityIntegrations", "Comment"),
+					g.KeywordOptions().SQL("SET"),
+				).
+				OptionalQueryStructField(
+					"Unset",
+					g.NewQueryStruct("AuthenticationPolicyUnset").
+						OptionalSQL("CLIENT_TYPES").
+						OptionalSQL("AUTHENTICATION_METHODS").
+						OptionalSQL("SECURITY_INTEGRATIONS").
+						OptionalSQL("MFA_AUTHENTICATION_METHODS").
+						OptionalSQL("MFA_ENROLLMENT").
+						OptionalSQL("COMMENT").
+						WithValidation(g.AtLeastOneValueSet, "ClientTypes", "AuthenticationMethods", "Comment", "SecurityIntegrations", "MfaAuthenticationMethods", "MfaEnrollment"),
+					g.ListOptions().NoParentheses().SQL("UNSET"),
+				).
+				Identifier("RenameTo", g.KindOfTPointer[SchemaObjectIdentifier](), g.IdentifierOptions().SQL("RENAME TO")).
+				WithValidation(g.ValidIdentifier, "name").
+				WithValidation(g.ExactlyOneValueSet, "Set", "Unset", "RenameTo").
+				WithValidation(g.ValidIdentifierIfSet, "RenameTo"),
+		).
+		DropOperation(
+			"https://docs.snowflake.com/en/sql-reference/sql/drop-authentication-policy",
+			g.NewQueryStruct("DropAuthenticationPolicy").
+				Drop().
+				SQL("AUTHENTICATION POLICY").
+				IfExists().
+				Name().
+				WithValidation(g.ValidIdentifier, "name"),
+		).
+		ShowOperation(
+			"https://docs.snowflake.com/en/sql-reference/sql/show-authentication-policies",
+			g.DbStruct("showAuthenticationPolicyDBRow").
+				Field("created_on", "string").
+				Field("name", "string").
+				Field("comment", "string").
+				Field("database_name", "string").
+				Field("schema_name", "string").
+				Field("owner", "string").
+				Field("owner_role_type", "string").
+				Field("options", "string"),
+			g.PlainStruct("AuthenticationPolicy").
+				Field("CreatedOn", "string").
+				Field("Name", "string").
+				Field("Comment", "string").
+				Field("DatabaseName", "string").
+				Field("SchemaName", "string").
+				Field("Owner", "string").
+				Field("OwnerRoleType", "string").
+				Field("Options", "string"),
+			g.NewQueryStruct("ShowAuthenticationPolicies").
+				Show().
+				SQL("AUTHENTICATION POLICIES").
+				OptionalLike().
+				OptionalIn().
+				OptionalStartsWith().
+				OptionalLimit(),
```

Comment:
The documentation does not mention those options as viable, but I checked that at least `like` works, please check others and test them in the integration test, because I believe they're not tested there. (I'll notify the doc team about it)

Reply:
Tested and posted to doc team